### ISSUE TITLE
aws-vault: error: exec: exec: "": executable file not found in %PATH%

### ISSUE THREAD

**Issue by cjannis** [99designs/aws-vault#891]
Here is the --debug output

aws-vault exec userprofile --debug
2022/03/15 09:00:46 aws-vault v6.5.0
2022/03/15 09:00:46 Loading config file C:\Users\username\.aws\config
2022/03/15 09:00:46 Parsing config file C:\Users\username\.aws\config
2022/03/15 09:00:46 [keyring] Considering backends: [wincred]
2022/03/15 09:00:47 Created new OIDC client (expires at: 2022-06-13 09:00:47 -0700 PDT)
2022/03/15 09:00:47 Created OIDC device code for https://ssoname.awsapps.com/start (expires in: 600s)
2022/03/15 09:00:47 Opening SSO authorization page in browser
Opening the SSO authorization page in your default browser (use Ctrl-C to abort)
https://device.sso.us-west-2.amazonaws.com/?user_code=####-####
2022/03/15 09:00:54 Created new OIDC access token for https://ssoappsname.com/start (expires in: 28800s)
2022/03/15 09:00:54 Got credentials ******************** for SSO role AWSAdministratorAccess (account: ##########), expires in 3h59m58.4069911s
2022/03/15 09:00:54 Setting subprocess env: AWS_DEFAULT_REGION=us-west-2, AWS_REGION=us-west-2
2022/03/15 09:00:54 Setting subprocess env: AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY
2022/03/15 09:00:54 Setting subprocess env: AWS_SESSION_TOKEN, AWS_SECURITY_TOKEN
2022/03/15 09:00:54 Setting subprocess env: AWS_SESSION_EXPIRATION
2022/03/15 09:00:54 Starting child process:
aws-vault: error: exec: exec: "": executable file not found in %PATH%

**Comment by acolominas** [comment]
Hi cjannis!
There is a workaround to fix this bug. You need to add "-- CMD.exe" at the end of the command.
e.g. aws-vault exec {my-profile} -- CMD.exe

**Comment by 0x4d4e** [comment]
Thanks for the workaround. Had the same issue running ssh in powershell from aws-vault.

**Comment by CreativeCreature7** [comment]
The workaround doesn't work for me. is there any way to add the executable to PATH?

**Comment by cande1gut** [comment]
@CreativeCreature7 in what environment are you trying? If is under Linux, you can pass `-- bash` and it should work.

**Comment by bob-walker-nhs** [comment]
Could anyone explain a bit more what's going on here and why the workaround works please? 
Is it possible to implement as a fix? 

I get the same issue with a devcontainer on WSL2, but with the same aws-vault binary, backend, and aws config on WSL2 **_without_** the devcontainer, I don't get the issue. 

_(I understand the purpose of `--`, but not the trailing `bash` or what it's telling aws-vault to do)._

**Comment by cande1gut** [comment]
@bob-walker-nhs is not a workaround, is more sort of a shortcut. Basically, when you call bash or cmd you are nesting a command line that will let you execute AWS commands directly while using the declared profile, compared to the basic usage where you need to type all the aws-vault command:

Instead of calling your commands in full:
```
1. aws-vault exec <profile> -- <aws-command1>
2. aws-vault exec <profile> -- <aws-command2>
...
```
You can do:
```
1. aws-vault exec <profile> -- <bash/cmd>
2. <aws-command1>
3. <aws-command2>
...
```

You can find more info in the [usage documentation](https://github.com/99designs/aws-vault/blob/master/USAGE.md), under "Executing a command".

**Comment by bob-walker-nhs** [comment]
Thanks for the clarification [cande1gut](https://github.com/cande1gut)!

I was struggling to understand why this happens in my dev container but not in a regular WSL2 shell, and I've found that it seems to be do with the interaction between VS Code, its terminal, and the Docker container. From the code, `exec.go` is using  `os.GetEnv('SHELL')` , but for reasons that escape me, `SHELL` is not set, despite the user's shell being set correctly in `/etc/passwd`, and available in a terminal shell:
```
$ echo $SHELL
/bin/bash
```
... but the `env` command (and presumably this is how `os.GetEnv` works?) does not:
```
$ env | grep SHELL
$
```
I've clumsily just added `export SHELL=/bin/bash` to the container user's `.bashrc` (and this seems to work), but I feel like I'm definitely missing a point here. I also don't want to add the `-- bash` workaround as this is for scripts that others use who currently have no issue with it.

**Comment by mtibben** [comment]
Fixed in #1136 + d169b7e38ee3dd9d7522149b7c4a6b5912463cda
